### ISSUE TITLE
No support for inequality requirements in python version dependency line

### ISSUE THREAD

**Issue by bertjwregeer** [sdispater/poetry#514]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: MacOS
- **Poetry version**: 0.12.2
- Included below

```
[tool.poetry.dependencies]
python = ">=3.7"
```

## Issue

The inequality requirements as documented here: https://poetry.eustace.io/docs/versions/#inequality-requirements should work for the python version too.

Right now installing something like `urllib3` doesn't work:

```
Riley:testing xistence$ poetry add urllib3
Using version ^1.24 for urllib3

Updating dependencies
Resolving dependencies... (0.2s)
                                                                                                            
[SolverProblemError]                                                                         
The current supported Python versions are >=3.7                                                           
Because no versions of urllib3 match >1.24,<2.0                                                           
 and urllib3 (1.24) requires Python >=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, <4, urllib3 is forbidden.  
So, because test depends on urllib3 (^1.24), version solving failed.                                      
                                                                                                            
add [-D|--dev] [--git GIT] [--path PATH] [-E|--extras EXTRAS] [--optional] [--python PYTHON] [--platform PLATFORM] [--allow-prereleases] [--dry-run] [--] <name> (<name>)...
```

**Comment by sdispater** [comment]
This is the expected behavior, and is due to one of the changes in the resolver.

Basically you python constraint has no upper bound so it means it's also compatible with Python >= 4.0 while `urllib3` is only compatible with Python <4 so the resolver cannot select it.

```python
>>> from poetry.semver import parse_constraint
>>> parse_constraint(">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, <4").allows_all(parse_constraint(">=3.7"))
False
```

To fix this you just have to change `>=3.7` to `^3.7`.

**Comment by bertjwregeer** [comment]
The rationale makes sense, just not something I was expecting. Thanks for the explanation!
